src/main/java/reactor/ipc/netty/channel/ServerContextHandler.java

```diff
@@ -102,17 +115,59 @@ public boolean isDisposed() {
 		         .isActive();
 	}
 
+	private void createAndRegisterTerminateEmitterIfAbsent(Channel ch) {
+		DirectProcessor<Void> onTerminateEmitter = DirectProcessor.create();
+		if(ch.attr(ON_TERMINATE_EMITTER).compareAndSet(null, onTerminateEmitter)) {
+			FutureMono.from(ch.closeFuture()).subscribe(new CoreSubscriber<Void>() {
+				@Override
+				public void onSubscribe(Subscription s) {
+					if (!ch.attr(CLOSE_FUTURE_SUBSCRIPTION).compareAndSet(null, s)) {
+						throw new IllegalStateException(
+								"A previous subscription was present.");
+					}
+					onTerminateEmitter.onSubscribe(s);
+				}
+
+				@Override
+				public void onNext(Void aVoid) {
+					onTerminateEmitter.onNext(aVoid);
+				}
+
+				@Override
+				public void onError(Throwable t) {
+					onTerminateEmitter.onError(t);
+				}
+
+				@Override
+				public void onComplete() {
+					onTerminateEmitter.onComplete();
+				}
+			});
```

Comment:
do we need to remove `CLOSE_FUTURE_SUBSCRIPTION` and `ON_TERMINATE_EMITTER ` when `onError` or `onComplete`?

Reply:
Yes especially if its a persistent connection to avoid leaking references back.